python.mk

```diff
@@ -37,107 +40,132 @@ ${PYTHON} ${ENV_DIR} ${PIP}:
 	${PIP} install --upgrade pip==9.0.1
 	${PIP} install wheel==0.30.0
 
-# Dummy targets onto which the targets defined in python_component below are
-# added as dependencies.
-${ENV_DIR}/.wheels-cleaned:
 	touch $@
 
-${ENV_DIR}/.wheels-built:
-	touch $@
+# This target builds all required wheelhouses and is therefore expected to be a
+# dependency of `make deb`
+.PHONY: wheelhouses
+
+# A dummy test target, to which dependencies are added for each test component
+.PHONY: test
 
-${ENV_DIR}/.wheels-installed:
+# Common coverage target. Because coverage is run over all components at once,
+# COVERAGE_SETUP_PY must be specified as a list of setup.py files that are run
+# under coverage.
+#
+# Optional parameters:
+#  - COVERAGE_SRC_DIR is the directory in which we run coverage.
+#  - COVERAGE_EXCL are files to be excluded from coverage
+#  - TEST_PYTHON_PATH is the python path to use
+#  - COMPILER_FLAGS are any additional flags to  be used.
```

Comment:
Should these be defaulted to anything use `?=`? If so, should the explanation be near that definition rather than here?

Reply:
I don't think so - that's making assumptions on the structure of the code that uses this makefile, which I've tried to avoid